ansible_runner/utils/streaming.py

```diff
@@ -85,6 +86,11 @@ def unstream_dir(stream, length, target_directory):
 
                 archive.extract(info.filename, path=target_directory)
 
+                # Fancy logic to preserve modification times
```

Comment:
Can we add a bit more context here to this comment as to _why_ we need to preserve modifications time?

Reply:
I thought this would be kind of a general expectation for extracting a zip file.

In the case of runner, we have a control & execution machine. The control machine creates the private_data_dir, the worker runs the job, and it gets streamed back. The modtimes can be used to distinguish if the control machine last modified the file, or if the file was modified during the job run.

In the AWX case, we use modification times to determine if new facts were written for a host. This is a big performance optimization for the case where the inventory has 10k hosts but you run against 1 host. Most playbooks will gather facts, and AWX maintains a fact cache in the database. A job run which should go super fast would get bogged down saving the rather substantial facts data structure for 10k hosts.

I have trouble putting that into a short sentence :)

Reply:
The new Executor team (currently just myself and @eqrx) lack A LOT of context around design decisions in the current code base (much of it having to do with AWX where we have no experience or insight), so it makes maintaining it rather difficult. Just a simple sentence like "AWX uses  modification times to determine if new facts were written for a host" would give us a lot of clarity when we see code like this where the purpose is not immediately clear.